app/controllers/structures_controller.rb

```diff
@@ -18,13 +18,18 @@ def show
   def update
     authorize! :manage_item, @cocina
 
-    status = StructureUpdater.from_csv(@cocina, params[:csv].read)
-
-    if status.success?
-      object_client.update(params: @cocina.new(structural: status.value!))
-      redirect_to solr_document_path(@cocina.externalIdentifier), status: :see_other, notice: 'Structural metadata updated'
+    state_service = StateService.new(@cocina.externalIdentifier, version: @cocina.version)
+    if state_service.allows_modification?
+      status = StructureUpdater.from_csv(@cocina, params[:csv].read)
+
+      if status.success?
+        object_client.update(params: @cocina.new(structural: status.value!))
+        redirect_to solr_document_path(@cocina.externalIdentifier), status: :see_other, notice: 'Structural metadata updated'
+      else
+        redirect_to solr_document_path(@cocina.externalIdentifier), flash: { error: status.failure.join('\n') }
+      end
     else
-      redirect_to solr_document_path(@cocina.externalIdentifier), flash: { error: status.failure.join('\n') }
+      redirect_to solr_document_path(@cocina.externalIdentifier), status: :not_acceptable, flash: { error: 'Updates not allowed on this object.' }
```

Comment:
I don't think you can redirect with a 406 status.

Reply:
yeah, i wasn't sure either, though the tests seemed to work this way... i put it back to a redirect and do other verification in the test to ensure we hit the correct code path